vizro-core/docs/pages/explanation/contributing.md

```diff
@@ -1,192 +1,159 @@
+    For either method, Hatch is the _only development dependency_. You do not need to manually install Python or create any virtual environments to develop Vizro; all this will be handled for you behind the scenes by Hatch. We have also configured our codespace to pre-install Hatch. If you develop on GitHub Codespaces you don't need to install anything at all!
 
-## Got a Vizro question?
 
-We are happy to receive general questions around Vizro. Take a look at our [issues page](https://github.com/mckinsey/vizro/issues) and raise a ticket in the category `general question`. We would be grateful if you could check for any similar descriptions in the existing issues before opening a new ticket.
+## Develop on GitHub Codespaces
 
-If you are already active on the Plotly Community Forum, you can also ask your question over there in the [Dash Python category](https://community.plotly.com/c/python/25).
+!!! note
 
-## How to interact with the repository
+    There is no need to manually create a fork of the Vizro code if you use GitHub Codespaces, because fork is [automatically created for you](https://docs.github.com/en/codespaces/developing-in-a-codespace/using-source-control-in-your-codespace#about-automatic-forking).
+    
+To develop on [GitHub Codespaces](https://docs.github.com/en/codespaces), follow the below steps:
 
-The easiest way to get up and running is to [open the repository in GitHub Codespaces](https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=626855142). This creates a temporary development environment with all the necessary configurations, making it especially convenient for tasks like reviewing pull requests.
+1. Use [Vizro's GitHub issues](https://github.com/mckinsey/vizro/issues) to discuss your contribution.
+2. [Create a codespace for our repository](https://codespaces.new/mckinsey/vizro). Leave the settings on their defaults and click "Create codespace" to start your codespace. It should take 1-2 minutes to fully launch and automatically start an example dashboard on port 8050.
+3. Make changes to Vizro code in your codespace. See the [GitHub Codespaces documentation on developing in a codespace](https://docs.github.com/en/codespaces/developing-in-a-codespace/developing-in-a-codespace) for more information.
+4. Add your name to the [list of contributors](authors.md) (source file `vizro-core/docs/pages/explanation/authors.md`).
+5. [Create a pull request (PR)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
+6. Before it can be merged, your PR must meet all the following requirements:
+    * Two approving reviews (including a code owner).
+    * Continuous Integration (CI) checks pass.
+    * Up-to-date with `main`.
```

Comment:
I'm tempted to take this out and put it up top so it applies to both methods of contributing, to avoid repetition.

Also, should you add something about the changelog requirement here too, and link through to the hatch command to add it?

Reply:
Should we also mention that some CI needs to be activated?

Reply:
@stichbury yeah, I wasn't sure about this. Given that a contributor would be using either the Codespace route or the local development route I thought it would be easier for someone to just follow all the steps in one of those sections without having to jump between a separate section that contains the parts that are common to both those processes and then back again. But it definitely makes sense to pull out point 1 on the list of both to the top of the page, so let me try and move this last point there as well and see if it reads ok.

I decided to not add anything about the changelog requirement here because:
* it's included in the "Continuous Integration (CI) checks pass" point
* when that check fails, the error message is very clear and easy to follow since it instructs the user to run `hatch run changelog:add`
* in theory changelogs aren't needed for every change anyway (purely docs changes shouldn't require it - if that's not the case then it's not working right)

@maxschulz-COL what do you mean by activated? Do the required jobs for external contributors not get triggered without some intervention?

Reply:
I was under the impression that yes. When I did my testing I wasn't focussing on that, but it looked to me like it. @l0uden do you know if there is any activation needed when raising a PR from a fork?

I can test later again though just to be sure

Reply:
1. Indeed seems like the automatic forking is not 100% reliable... but if it's working for you now then agreed with your proposal - let's leave the contribution guide as is then 👍 
2. Ah yes, the reason for this is this setting (which we currently have to be as permissive as possible). I will add a mention to this to the guide, good spot!
![image](https://github.com/user-attachments/assets/d1b3b18f-c350-44a7-b975-b5a14d72592e)

Reply:
@maxschulz-COL this is well worth knowing for GHC actually, given that many of them might have new github accounts so fit into this category.